src/compiler/checker.ts

```diff
@@ -9558,23 +9851,50 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
 
             function serializeEnum(symbol: Symbol, symbolName: string, modifierFlags: ModifierFlags) {
+                const internalSymbolName = getInternalSymbolName(symbol, symbolName);
+                context.approximateLength += 9 + internalSymbolName.length; // `enum internalName { }`
+                const members: EnumMember[] = [];
+                const memberProps = filter(getPropertiesOfType(getTypeOfSymbol(symbol)), p => !!(p.flags & SymbolFlags.EnumMember));
+                let i = 0;
+                for (const p of memberProps) {
+                    i++;
+                    if (checkUnfoldingTruncationLength(context) && (i + 2 < memberProps.length - 1)) {
+                        context.out.truncated = true;
+                        members.push(factory.createEnumMember(` ... ${memberProps.length - i} more ... `));
+                        const last = memberProps[memberProps.length - 1];
+                        const initializedValue = last.declarations && last.declarations[0] && isEnumMember(last.declarations[0]) ? getConstantValue(last.declarations[0]) : undefined;
+                        const initializer = initializedValue === undefined ? undefined :
+                            typeof initializedValue === "string" ? factory.createStringLiteral(initializedValue) :
+                            factory.createNumericLiteral(initializedValue);
+                        const memberName = unescapeLeadingUnderscores(last.escapedName);
+                        const member = factory.createEnumMember(
+                            memberName,
+                            initializer,
+                        );
+                        members.push(member);
+                        break;
+                    }
+                    // TODO: Handle computed names
+                    // I hate that to get the initialized value we need to walk back to the declarations here; but there's no
+                    // other way to get the possible const value of an enum member that I'm aware of, as the value is cached

```

Comment:
The intent here to seems to be show the enum's calculated value as its' initializer, even if said initialized value isn't an important part of the type. For non-literal enums, the calculated value is possibly less important than just the initializer as-written (and probably carries less documenting power). If you only want to show calculated values on enums where the value is part of the type, you can just get the `.value` of the literal type that `getTypeOfSymbol(member)` returns. If the desire is to show more for non-literal enums, it might be better to use the initializer node on the declaration verbatim, rather than showing a calculated result that may or may not actually affect how we interpret the type because another member was non-literal.

Reply:
I updated this to use the original initializer, if present. I don't think I need to be very concerned with the initializer referring to things that are out of scope, etc, since this is just for a hover, but I wasn't sure how to handle the initializer node reuse. Let me know if I need to do something more fancy than just deep cloning.